internal/backend/remote-state/kubernetes/backend.go

```diff
@@ -49,7 +51,22 @@ func New() backend.Backend {
 			"secret_suffix": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Suffix used when creating the secret. The secret will be named in the format: `tfstate-{workspace}-{secret_suffix}`.",
+				Description: "Suffix used when creating the secret. The secret will be named in the format: `tfstate-{workspace}-{secret_suffix}`. Note that the backend may append its own numeric index to the secret name when chunking large state files into multiple secrets. In this case, there will be multiple secrets named in the format: `tfstate-{workspace}-{secret_suffix}-{index}`.",
+				ValidateFunc: func(v interface{}, s string) ([]string, []error) {
+					value := v.(string)
+					// Split the suffix by '-' and check the last segment.
+					parts := strings.Split(value, "-")
+					if len(parts) > 1 {
+						lastPart := parts[len(parts)-1]
+						if _, err := strconv.Atoi(lastPart); err == nil {
+							// If the last segment is a number, it's considered invalid.
+							// The backend automatically appends its own numeric suffix when chunking large state files into multiple secrets.
+							// Allowing a user-defined numeric suffix could cause conflicts with this mechanism.
+							return nil, []error{fmt.Errorf("secret_suffix must not end with '-<number>', got '%s'", value)}
```

Comment:
Just a nit but you can use the `%q` directive to double quote strings instead of doing it yourself. In other places we've tended to use double quotes in error messages.

Reply:
Fixed